mirroring/mirroring.go

```diff
@@ -119,42 +114,41 @@ func VerifySignedTreeHead(logInfo *models.LogInfo, pubkey string) error {
 	return nil
 }
 
-func VerifyLogConsistency(rekorClient *gclient.Rekor, oldSize int64, oldRootHash string) (int64, string, error) {
+func VerifyLogConsistency(rekorClient *gclient.Rekor, oldSize int64, oldRootHash []byte) (*util.SignedCheckpoint, error) {
 	logInfo, err := GetLogInfo(rekorClient)
 	if err != nil {
-		return 0, "", err
+		return nil, err
 	}
 
 	logProof, err := GetLogProof(rekorClient, &oldSize, logInfo.TreeSize)
 	if err != nil {
-		return 0, "", err
-	}
-
-	oldRoot, err := hex.DecodeString(oldRootHash)
-	if err != nil {
-		return 0, "", err
+		return nil, err
 	}
 
 	newRoot, err := hex.DecodeString(*logInfo.RootHash)
 	if err != nil {
-		return 0, "", err
+		return nil, err
 	}
 
 	proofs := make([][]byte, len(logProof.Hashes))
 	for i, h := range logProof.Hashes {
 		hash, err := hex.DecodeString(h)
 		if err != nil {
-			return 0, "", err
+			return nil, err
 		}
 		proofs[i] = hash
 	}
 
 	verifier := logverifier.New(rfc6962.DefaultHasher)
-	err = verifier.VerifyConsistencyProof(oldSize, *logInfo.TreeSize, oldRoot, newRoot, proofs)
+	err = verifier.VerifyConsistencyProof(oldSize, *logInfo.TreeSize, oldRootHash, newRoot, proofs)
```

Comment:
note: this is deprecated https://github.com/sigstore/rekor/pull/807

Reply:
Updated to the latest rekor and go 1.19, and fixed